roles/reproducer/tasks/reuse_main.yaml

```diff
@@ -0,0 +1,274 @@
+  tags:
+    - always
+  vars:
+    default_path: >-
+      {{
+        cifmw_reproducer_default_repositories |
+        selectattr('src', 'match', '^.*/ci[_\-]framework$') |
+        map(attribute='dest') | first
+      }}
+    custom_path: >-
+      {{
+        cifmw_reproducer_repositories |
+        selectattr('src', 'match', '^.*/ci-framework$') |
+        map(attribute='dest')
+      }}
+    _path: >-
+      {{
+        (custom_path | length > 0) |
+        ternary(custom_path | first, default_path)
+      }}
+  ansible.builtin.set_fact:
+    _cifmw_reproducer_framework_location: >-
+      {{
+        (_path is match('.*/ci-framework/?$')) |
+        ternary(_path, [_path, 'ci-framework'] | path_join)
+      }}
+
+- name: Set _use_crc based on actual layout
+  tags:
+    - always
+  vars:
+    _use_crc: >-
+      {{
+        _cifmw_libvirt_manager_layout.vms.crc is defined and
+        (
+          (_cifmw_libvirt_manager_layout.vms.crc.amount is defined and
+           _cifmw_libvirt_manager_layout.vms.crc.amount|int > 0) or
+        _cifmw_libvirt_manager_layout.vms.crc.amount is undefined)
+      }}
+    _use_ocp: >-
+      {{
+        _cifmw_libvirt_manager_layout.vms.ocp is defined and
+        (_cifmw_libvirt_manager_layout.vms.ocp.amount is defined and
+         _cifmw_libvirt_manager_layout.vms.ocp.amount|int > 0)
+      }}
+  ansible.builtin.set_fact:
+    _use_crc: "{{ _use_crc }}"
+    _use_ocp: "{{ _use_ocp }}"
+    _has_openshift: "{{ _use_ocp or _use_crc }}"
+
+- name: Ensure directories are present
+  tags:
+    - always
+  ansible.builtin.file:
+    path: "{{ cifmw_reproducer_basedir }}/{{ item }}"
+    state: directory
+    mode: "0755"
+  loop:
+    - artifacts
+    - logs
+
+- name: Load the architecture local kustomize patches
+  when:
+    - cifmw_architecture_scenario is defined
+  ansible.builtin.include_role:
+    name: kustomize_deploy
+    tasks_from: generate_base64_patches_from_tree.yml
+
+- name: Run only on hypervisor with controller-0
+  block:
+    - name: Push local code
+      ansible.builtin.include_tasks: push_code.yml
+
+    - name: Group tasks on controller-0
+      delegate_to: controller-0
+      block:
+        - name: Inject CI Framework motd
+          become: true
+          ansible.builtin.template:
+            dest: "/etc/motd.d/cifmw.motd"
+            src: "motd.j2"
+            mode: "0644"
+
+        - name: Rotate some logs
+          tags:
+            - always
+          ansible.builtin.include_tasks: rotate_log.yml
+          loop:
+            - "/home/zuul/ansible-bootstrap.log"
+
+        - name: Bootstrap environment on controller-0
+          environment:
+            ANSIBLE_LOG_PATH: "~/ansible-bootstrap.log"
+          no_log: "{{ cifmw_nolog | default(true) | bool }}"
+          ansible.builtin.command:
+            chdir: "{{ _cifmw_reproducer_framework_location }}"
+            cmd: >-
+              ansible-playbook -i ~/ci-framework-data/artifacts/zuul_inventory.yml
```

Comment:
FYI, we're trying to get rid of such nested invocations of playbooks. We're not there yet, so we cannot do anything about that. However, we're gonna be slowly retiring such a dirty workarounds.

Reply:
Generally I'm against this as well I would follow you as you make those changes to remove them here as well